components/server/src/main/java/com/hotels/styx/server/handlers/ClassPathResourceHandler.java

```diff
@@ -33,32 +34,54 @@
 import static com.hotels.styx.api.HttpResponseStatus.OK;
 import static com.hotels.styx.server.handlers.MediaTypes.mediaTypeOf;
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+import static java.util.regex.Matcher.quoteReplacement;
 
 /**
  * Handler for class path resources.
  */
 public class ClassPathResourceHandler extends BaseHttpHandler {
+    private static final Pattern DOUBLE_SEPARATOR = Pattern.compile("//", Pattern.LITERAL);
+    private final String allowedPrefix;
     private final String root;
 
     public ClassPathResourceHandler(String root) {
-        this.root = ensureRootEndsInSlash(root);
+        this(root, root);
     }
 
-    private static String ensureRootEndsInSlash(String root) {
-        return root.endsWith("/") ? root : root + "/";
+    public ClassPathResourceHandler(String allowedPrefix, String root) {
```

Comment:
Perhaps call these parameters `requestPrefix` and `resourceRoot`? To make it a little clearer what the strings get applied to/checked against.